src/main/java/uk/gov/gchq/palisade/rule/Rules.java

```diff
@@ -165,58 +166,55 @@ public String getMessage() {
      * @return {@code true} if this rule set contains at least one rule
      */
     public boolean containsRules() {
-        return !rules.isEmpty();
+        return !rulesHashMap.isEmpty();
     }
 
     @Override
     public boolean equals(final Object o) {
         boolean rtn = (this == o);
-        if (!rtn) {
-            if (o != null && this.getClass() == o.getClass()) {
-
-                final Rules<?> that = (Rules<?>) o;
-
-                final EqualsBuilder builder = new EqualsBuilder()
-                        .append(message, that.message)
-                        .append(this.rules.keySet(), that.getRules().keySet());
-
-                if (builder.isEquals()) {
-                    for (final Entry<String, Rule<T>> entry : this.rules.entrySet()) {
-                        final String ruleName = entry.getKey();
-                        final Rule thisRule = entry.getValue();
-                        final Rule thatRule = that.getRules().get(ruleName);
-
-                        builder.append(thisRule.getClass(), thatRule.getClass());
-                        if (builder.isEquals()) {
-                            // This is expensive - but we don't have any other way of doing it
-                            builder.append(JSONSerialiser.serialise(thisRule), JSONSerialiser.serialise(thatRule));
-                        }
-
-                        if (!builder.isEquals()) {
-                            break;
-                        }
+        if (!rtn && (o != null && this.getClass() == o.getClass())) {
+
+            final Rules<?> that = (Rules<?>) o;
+
+            final EqualsBuilder builder = new EqualsBuilder()
+                    .append(message, that.message)
+                    .append(this.rulesHashMap.keySet(), that.getRules().keySet());
+
+            if (builder.isEquals()) {
+                for (final Entry<String, Rule<T>> entry : this.rulesHashMap.entrySet()) {
+                    final String ruleName = entry.getKey();
+                    final Rule thisRule = entry.getValue();
+                    final Rule thatRule = that.getRules().get(ruleName);
+
+                    builder.append(thisRule.getClass(), thatRule.getClass());
+                    if (builder.isEquals()) {
+                        // This is expensive - but we don't have any other way of doing it
+                        builder.append(JSONSerialiser.serialise(thisRule), JSONSerialiser.serialise(thatRule));
+                    }
+
+                    if (!builder.isEquals()) {
+                        break;
```

Comment:
can return false at this point as we have already determined that there is not a match and there is, therefore no need to call the isEquals() on line 200.